nemo/collections/asr/metrics/wer_bpe.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+import editdistance
+import torch
+from pytorch_lightning.metrics import TensorMetric
+
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+
+
+class WERBPE(TensorMetric):
```

Comment:
I really encourage everyone adding new metric to spend time and write a unit test for it. This will save you from a lot of suffering in the future :)

Reply:
I was planning to, but theres multiple issues - need the AN4 dataset, need to run the tokenizer on it (2 different kinds), store them in temporary locations, run two tests on them and then delete the tokenizers. I can still test WPE based tokenizers by hardcoding the tokenizer configs into code and then temp writing them to files, but SentencePiece cant be tested that way as that library generates a tokenizer.model binary file for us along with a vocab.

Reply:
currently, WERBPE is almost entirely based off of WER, with only 3 lines of change - takes tokenizers instead of labels, and does self.tokenizer.ids_to_text() instead of mapping int ids to tokens from a dictionary. It should be fairly safe. I do want to add tests, but we need a discussion on how to store test data again.